lib/dry/web/settings.rb

```diff
@@ -0,0 +1,59 @@
+require "yaml"
+
+module Dry
+  module Web
+    class Settings
+      TypeError = Class.new(StandardError)
+
+      AnyType = Class.new do
+        def self.[](value)
+          value
+        end
+      end
```

Comment:
I've replaced `AnyType` with `nil` and an `if` clause around type coercion. Less code :)